src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,272 @@
+        }
+
+        return taskList;
+    }
+
+    /**
+     * Marks a task as complete in the storage file by updating its status from incomplete
+     * (0) to complete (1).
+     *
+     * @param t The task to be marked as complete.
+     */
+    public void markComplete(Task t) {
+        try {
+            List<String> lines = Files.readAllLines(Paths.get(this.filePath));
+            List<String> updatedLines = lines.stream().map(line -> {
+                String[] parts = line.split("\\|");
+                if (parts.length > 2 && parts[2].trim().equals(t.getName())) {
+                    parts[1] = " 1 "; // Change the status from 0 to 1
+                    return String.join("|", parts).trim();
+                }
+                return line;
+            }).toList();
+
+            // Write the updated lines back to the file
+            try (FileWriter writer = new FileWriter(this.filePath, false)) {
+                for (String updatedLine : updatedLines) {
+                    writer.write(updatedLine + System.lineSeparator());
+                }
+            }
+
+        } catch (IOException e) {
+            System.out.println("An error occurred while writing to the file.");
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Marks a task as incomplete in the storage file by updating its status from complete
+     * (1) to incomplete (0).
+     *
+     * @param t The task to be marked as incomplete.
+     */
+    public void markIncomplete(Task t) {
+        try {
+            List<String> lines = Files.readAllLines(Paths.get(this.filePath));
+            List<String> updatedLines = lines.stream().map(line -> {
+                String[] parts = line.split("\\|");
+                if (parts.length > 2 && parts[2].trim().equals(t.getName())) {
+                    parts[1] = " 0 "; // Change the status from 1 to 0
+                    return String.join("|", parts).trim();
+                }
+                return line;
+            }).toList();
+
+            // Write the updated lines back to the file
+            try (FileWriter writer = new FileWriter(this.filePath, false)) {
+                for (String updatedLine : updatedLines) {
+                    writer.write(updatedLine + System.lineSeparator());
+                }
+            }
+
+        } catch (IOException e) {
+            System.out.println("An error occurred while writing to the file.");
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Deletes a task from the storage file by removing the corresponding line from the file.
+     *
+     * @param t The task to be deleted from the storage file.
+     */
+    public void deleteTask(Task t) {
+        try {
+            // Read all lines from the file
+            List<String> lines = Files.readAllLines(Paths.get(this.filePath));
+
+            // Filter out the line corresponding to the task to be deleted
```

Comment:
I like how you add comments at each section of the code to explain what it is doing